lib/ssl/src/ssl_config.erl

```diff
@@ -2179,24 +2174,32 @@ do_handle_cert_file(File, PemCacheName) ->
 handle_key_file(#{keyfile := File} = CertKey, PemCacheName) ->
     case file:read_file(File) of
         {ok, Pem} ->
-            case public_key:pem_decode(Pem) of
+            PemEntries = public_key:pem_decode(Pem),
+            Password = maps:get(password, CertKey, ""),
+            case key_entry(PemEntries) of
                 [KeyEntry] ->
-                    Password = maps:get(password, CertKey, ""),
                     try public_key:pem_entry_decode(KeyEntry, Password) of
                         Key ->
                             handle_key(PemCacheName, File, Key, [KeyEntry])
```

Comment:
```
handle_key(PemCacheName, File, Key, PemEntries)
```
is needed here; insertion of only the `KeyEntry` into the `pem_cache` will lead to `{error, {options, {certfile, ..., no_certs}}}` since the existing entry with cert and key is overwritten.